eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/ExternalDependencyExposedCheck.java

```diff
@@ -0,0 +1,201 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.tools.checkstyle.checks;
+
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.FullIdent;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ *  No external dependency exposed in the public API
+ */
+public class ExternalDependencyExposedCheck extends AbstractCheck {
+    private static final String JAVA = "java";
+    private static final String COM_AZURE = "com.azure";
+    private static final String REACTOR = "reactor";
+    private static final String DOT = ".";
+
+    private static final String EXTERNAL_DEPENDENCY_ERROR =
+        "Class ''%s'', is a class from external dependency. You should not use it as a return or parameter argument type";
+
+    private static final Set<String> validLibrarySet = new HashSet<>();
+    private static Map<String, String> classPathMap = new HashMap<>();
+
+    @Override
+    public void init() {
+        validLibrarySet.add(JAVA);
```

Comment:
Every single time init is called, these will be added to the instance. It would be better to do this in the constructor, so these items are only ever added once.